src/main/groovy/com/github/rodm/teamcity/SignConfiguration.groovy

```diff
@@ -0,0 +1,46 @@
+package com.github.rodm.teamcity
+
+import org.gradle.api.Project
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.jetbrains.zip.signer.signer.CertificateUtils
+import org.jetbrains.zip.signer.signer.PrivateKeyUtils
+
+import java.security.PrivateKey
+import java.security.cert.X509Certificate
+
+class SignConfiguration {
+    private ListProperty<X509Certificate> certificateChain
+    private Property<PrivateKey> privateKey
+    private Project project
+
+    SignConfiguration(Project project) {
+        this.project = project
+        this.certificateChain = project.objects.listProperty(X509Certificate)
+        this.privateKey = project.objects.property(PrivateKey)
+    }
+
+    ListProperty<X509Certificate> getCertificateChain() {
+        return certificateChain
+    }
+
+    void setCertificateChain(List<X509Certificate> certificateChain) {
+        this.certificateChain.set(certificateChain)
+    }
+
+    Property<PrivateKey> getPrivateKey() {
+        return privateKey
+    }
+
+    void setPrivateKey(PrivateKey privateKey) {
+        this.privateKey.set(privateKey)
+    }
+
+    List<X509Certificate> loadCertificateChain(File file) {
+        return CertificateUtils.loadCertificatesFromFile(file)
+    }
+
+    PrivateKey loadPrivateKey(File file, String password = null) {
+        return PrivateKeyUtils.loadPrivateKey(file, password?.toCharArray())
+    }
+}
```

Comment:
@satamas the 2 methods loadCertificateChain and loadPrivateKey don't appear to be used anywhere?

Reply:
I've designed them as helper methods. Using those methods plugin developer can easily load their certificate and private key from the filesystem.

Reply:
I've looked at this again and think it would be better to have the certificate file, private key file and password as File and String properties on the SignConfiguration and SignPluginTask and move the helper methods to the task, what do you think?

Also do you a set of steps to create a certifacte file and password file for testing?

Reply:
> I've looked at this again and think it would be better to have the certificate file, private key file and password as File and String properties on the SignConfiguration and SignPluginTask and move the helper methods to the task, what do you think?

I forgot to answer this question, but that wasn't a good idea. There was a reason why I did it the way I did - a certificate can be loaded not from a file but from a string or from some certificate storage. And it will be very important for a CI tools  where it is a problem to create a file but it's easy to pass string via environment variables

> Also do you a set of steps to create a certifacte file and password file for testing?

There is a [bouncy castle](https://www.bouncycastle.org/) library in dependencies that can be used to generate test certificates and private keys